experimental/InvariantTheory/iterators.jl

```diff
@@ -91,8 +93,81 @@ function dimension_via_molien_series(::Type{T}, R::InvRing, d::Int) where T <: I
   return T(numerator(k))::T
 end
 
-# Iterate over the basis of the degree d component of an invariant ring.
-# algo can be either :default, :reynolds or :linear_algebra.
+@doc Markdown.doc"""
+     iterate_basis(IR::InvRing, d::Int, algo::Symbol = :default)
+
+Given an invariant ring `IR` and an integer `d`, return an iterator over a basis for the invariants in degree `d`.
+The used algorithm can be specified using the optional argument `algo`. Possible values are `:reynolds` which uses the reynolds operator to construct the basis (only available in the non-modular case) and `:linear_algebra` which uses plain linear algebra. With the default value `:default` the heuristically best algorithm is selected.
+When using the reynolds operator the basis is constructed element-by-element. With linear algebra this is not possible and the whole basis will be constructed directly when calling the function.
+
+See also `basis`.
+
+# Examples
+```
+julia> K, a = CyclotomicField(3, "a")
+(Cyclotomic field of order 3, a)
+
+julia> M1 = matrix(K, [0 0 1; 1 0 0; 0 1 0])
+[0   0   1]
+[1   0   0]
+[0   1   0]
+
+julia> M2 = matrix(K, [1 0 0; 0 a 0; 0 0 -a-1])
+[1   0        0]
+[0   a        0]
+[0   0   -a - 1]
+
+julia> G = MatrixGroup(3, K, [ M1, M2 ])
+Matrix group of degree 3 over Cyclotomic field of order 3
+
+julia> IR = invariant_ring(G)
+Invariant ring of
+Matrix group of degree 3 over Cyclotomic field of order 3
+with generators
+AbstractAlgebra.Generic.MatSpaceElem{nf_elem}[[0 0 1; 1 0 0; 0 1 0], [1 0 0; 0 a 0; 0 0 -a-1]]
+
+julia> B = iterate_basis(IR, 6)
+Iterator over a basis of the component of degree 6 of
+Invariant ring of
+Matrix group of degree 3 over Cyclotomic field of order 3
+with generators
+AbstractAlgebra.Generic.MatSpaceElem{nf_elem}[[0 0 1; 1 0 0; 0 1 0], [1 0 0; 0 a 0; 0 0 -a-1]]
+
+julia> collect(B)
+4-element Vector{MPolyElem_dec{nf_elem, AbstractAlgebra.Generic.MPoly{nf_elem}}}:
+ x[1]^2*x[2]^2*x[3]^2
+ x[1]^4*x[2]*x[3] + x[1]*x[2]^4*x[3] + x[1]*x[2]*x[3]^4
+ x[1]^3*x[2]^3 + x[1]^3*x[3]^3 + x[2]^3*x[3]^3
+ x[1]^6 + x[2]^6 + x[3]^6
+```
+```
```

Comment:
What's this about? Are those two different examples? Then I'd expect some text in between.